lld/wasm/Config.h

```diff
@@ -32,6 +32,7 @@ class InputTable;
 class InputGlobal;
 class InputFunction;
 class Symbol;
+struct WasmSym;
```

Comment:
Is this forward declaration needed?  I'd be surprised if `WasmSym sym{};` would work below with only a forward decl.

Reply:
Hmmm, I remember getting some error when I was building llvm without it locally :(

I can try finding the error !

Reply:
Hopefully we're correct in keeping WasmSym in symbols.h and just make use of it in Config.h (unlike the ELF case where even the struct is made in ELF/Config.h) 

I guess that might not be required here ?

Reply:
Okay I think it was best to maintain some sort of consistency with what ELF was doing.

We were doing the same here with regards to everything except maintaining WasmSym in config.h (just like elf does with maintaining ElfSym in config.h)

cc @sbc100 I have made the change as per this commit https://github.com/llvm/llvm-project/commit/03be619d9434de0a9616660a2119675635239a5b#diff-ba5e9bcbcabe54e23f58f9466afb49d9720373eb11b9b3a42ddc2d653b035594

which basically does the exact same change for ELF